sdk/core/core/src/az_span.c

```diff
@@ -153,6 +154,30 @@ AZ_NODISCARD az_result az_span_atou32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_atod(az_span span, az_span* conversion_buffer, double* out_number)
+{
+  AZ_PRECONDITION_VALID_SPAN(span, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_number);
+
+  if (conversion_buffer == NULL)
+  {
+    // This means there is already an invalid character after the last digit to convert
```

Comment:
I don't think it is worth it to validate that.
We would basically need to scan all span looking for a non-digit character after the first digit is found.

If conversion_buffer is null and there is not any non-digit character, we would have an unexpected behavior. If the next memory address is valid and accesible and is not a digit, function will still works.
But it could be also a segmentation fault.

However, this is C, right?  We are only adding a utility function on top of atof to customers to use az_span.